File: codeforces/practices/rating/1400/279B-Books.cpp
// https://codeforces.com/problemset/problem/279/B

#include <iostream>
#include <cstdint>
#include <cstdlib>
#include <cmath>
#include <array>
#include <vector>
#include <algorithm>
#define sz(a) ((int)((a).size()))
#define char unsigned char

using namespace std;

typedef long long ll;
typedef long double ld;

int solve() {
	int n,t;
	cin >> n >> t;
	int a[n];
	for (int i = 0; i < n; ++i)
	{
		cin >> a[i];
	}
	int ans = 0, sum = 0;
	int i = 0, j = 0;
	while(j < n && i < n) { 
		while (sum + a[j] <= t && j < n) {
			sum += a[j];
			j++;
		}
		ans = max(ans, j-i);
		sum += a[j];
		j++;
		while(sum > t) {
			sum -= a[i];
			i++;
		}
	}
	cout << ans << endl;
	return 0;
}

int32_t main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	solve();
}

File: codeforces/practices/rating/1400/279B-Books.py
# https://codeforces.com/problemset/problem/279/B

n, t = input().split()
t = int(t)
n = int(n)
nums = input().split()
nums = [int(x) for x in nums]

s, b = 0, 0
i, j = 0, 0
while (i < n) and (j < n):
    while j < n and s + nums[j] <= t:
        s += nums[j]
        j += 1
    b = max(b, j-i)
    if j >= n: break
    s += nums[j]
    j += 1
    while s > t:
        s -= nums[i]
        i += 1
print(b)